plat/nvidia/tegra/common/tegra_bl31_setup.c

```diff
@@ -379,13 +379,12 @@ void bl31_plat_arch_setup(void)
  ******************************************************************************/
 int bl31_check_ns_address(uint64_t base, uint64_t size_in_bytes)
 {
-	uint64_t end = base + size_in_bytes - 1;
+	uint64_t end = base + size_in_bytes;
 
 	/*
 	 * Check if the NS DRAM address is valid
 	 */
-	if ((base < TEGRA_DRAM_BASE) || (end > TEGRA_DRAM_END) ||
-	    (base >= end)) {
+	if ((base < TEGRA_DRAM_BASE) || (end > TEGRA_DRAM_END)) {
```

Comment:
Just to check... 'end' points to the address one after size_in_bytes starting from base; assume that TEGRA_DRAM_END also points to the address one after the last available / valid address?